src/components/TokenAllowanceGuard/TokenAllowanceGuard.tsx

```diff
@@ -52,11 +53,12 @@ const StyledAllowanceGuard = styled("div")(({ theme }) => ({
 }));
 
 export const TokenAllowanceGuard: React.FC<{
+  balance: DecimalBigNumber;
   message: ReactNode;
   isVertical?: boolean;
   tokenAddressMap: AddressMap;
   spenderAddressMap: AddressMap;
-}> = ({ message, isVertical = false, tokenAddressMap, spenderAddressMap, children }) => {
+}> = ({ balance, message, isVertical = false, tokenAddressMap, spenderAddressMap, children }) => {
   const approveMutation = useApproveToken(tokenAddressMap, spenderAddressMap);
   const { data: allowance } = useContractAllowance(tokenAddressMap, spenderAddressMap);
```

Comment:
rather than passing in `balance` as a param can you just query balance from `tokenAddressMap` from within TokenAllowanceGuard similar to how we query allowance?

Reply:
yep, just changed